### PR TITLE
Now with wheat field vectors

### PR DESCRIPTION
**Pull request by jordansread**
Current status PR. Leaving for a week

### CODE REVIEW

**Comment by jordansread** [comment]
image

Things that would make this better

 - add state outlines
 - better legend
 - better use of circle color ramp and better placement of colors on this to line up with "hot-shift"
 - mess around with resampling hex size, line widths, and x/y_scale factors to make it right amount of busy/cool
 - some way to encode the length of the vector? right now the color is the angle, but the total amount of heat/time shift is likely key emphasis

**Comment by jordansread** [comment]
@lindsayplatt data.table code reduced processing time from 6hrs to <6min!

**Comment by jordansread** [comment]
Perhaps something like this (legend and other info) would be helpful? - from ppt sketch: image

**Comment by cnell-usgs** [comment]
😍 😍 😍

File: 03_historical_jread/src/plot_utils.R [hunk 1]
```diff
@@ -1,12 +1,51 @@
...
+  # access Contiguous U.S. state map:
+  US_states <- sf::st_transform(spData::us_states, st_crs(early_data))
 
   ggplot() +
-    geom_sf(data = del_data, size = 0.5, aes(col=temp_diff)) +
-    scale_colour_viridis_c(option = "magma")
+    geom_sf(data = US_states, color = 'grey90', fill = 'white', size = 0.75) +
+    geom_segment(del_data, mapping = aes(col = angle, x = x, y = y, xend = xend, yend = yend),
+                 size = lwd, show.legend = FALSE) +
+    scico::scale_color_scico(palette = "romaO", direction = -1, midpoint = 0) +
```
Comment (cnell-usgs) [inline 03_historical_jread/src/plot_utils.R]: `midpoint = 0` is throwing an error, I don't think it's needed here

**Comment by cnell-usgs** [comment]
This is awesome! Resampling to the hex grid was really smart and I love the end result.

The way the color breaks line up in the legend is a little distracting since the circle is split into quadrants but there's an uneven number of color breaks. Not sure how to add the inner part of the legend you have on your sketch, but I [sent a PR to your fork that recreates the outer ring component ](https://github.com/jread-usgs/chart-challenge-22/pull/1) and adjusts the color breaks. 

Before posting, this needs and informative title (including dates of comparison) and explanation of what warmer and later means in terms of lakes. A lake getting warmer is intuitive but the timing part needs a little more context, this could be folded into the legend. I'd also suggest making the USGS logo a little more transparent so it doesn't overpower the map. Happy to compose the final viz in illustrator with you if that is helpful.

**Comment by jordansread** [comment]
Thanks for the tips - I merged your PR but I'm not used to building a single output image based on different images using ggplot2. For this, that would be the map, the USGS logo, and the compass rose. Is there a package you use to combine all of these files into one or do you typically do that in Illustrator? I've done this before with `graphics::` but not ggplot2.

**Comment by cnell-usgs** [comment]
In R I would typically use `patchwork` or `cowplot` to piece together different `ggplot` components. `cowplot` is better for adding a logo https://wilkelab.org/cowplot/articles/drawing_with_on_plots.html 

But if it's a one-off thing I like to pull into illustrator to be able to clean it up a little bit more.

**Comment by jordansread** [comment]
@ceenell I've got another stitched image in here - I'd love to make it all in on set of code but think I'm not going to have time to complete that. 

But I have added the title and legend edits, and explain the methods in the tweets that will accompany this. I could add more about the timing calculation in this subtitle, but it was challenging to explain in a concise way without crowding things out. (from the tweet thread text: *Day of year of hitting half of the yearly GDD value was the timing metric between the two decades*)

Added a README.md to this PR too. 

220403_historical_compare

File: 03_historical_jread/src/plot_utils.R (hunk 1 above)
Comment (cnell-usgs) [reply 03_historical_jread/src/plot_utils.R]: I was wrong about this, needed to update my package

**Comment by cnell-usgs** [comment]
Looks great.

**Comment by cnell-usgs** [comment]
@jread-usgs 

> @ceenell I've got another stitched image in hear - I'd love to make it all in on set of code but think I'm not going to have time to complete that.
> 
> But I have added the title and legend edits, and explain the methods in the tweets that will accompany this. I could add more about the timing calculation in this subtitle, but it was challenging to explain in a concise way without crowding things out. (from the tweet thread text: _Day of year of hitting half of the yearly GDD value was the timing metric between the two decades_)
> 
> Added a README.md to this PR too.
> 
> 220403_historical_compare

For future reference, there's a good example of how to piece together plot components with the USGS logo using `cowplot` here https://github.com/USGS-VIZLAB/chart-challenge-22/blob/141f7dc910342a7628c5314702a8c6fe7551bd99/19_shamshaw/src/plot_functions.R#L79-L98

**Pull request merged**